13 implement predictprovoc method

danerkestey opened the pull request:
Since no `provoc_obj` currently, the predict function assumes the `provoc_obj` has `proportions` and `variant_matrix` elements.

The function then takes those elements and performs matrix multiplication for predictions and returns the results in the same order.

danerkestey committed:
Skeleton for predict function

Created skeleton of predict function to be implemented

danerkestey committed:
Current skeleton of predict

danerkestey committed:
added check for predict

Added check to make sure the inputted provoc_obj has proportions and variant_matrix elements

DBecker7 reviewed (changes requested):
A few requests:

- The function should replicate the behaviour of [`predict.glm`](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/predict.glm). For now, all arguments as `NULL` with description "Not yet implemented". In the future, a `newdata` argument will be implemented to allow for predictions of other mutations that weren't observed in the data but exist in `variant_matrix`.
- The function should do some error checking to at least ensure the dimensions match. In the future, we'll ensure that the dimnames match, but we need to ensure define other functions so that `proportions` has dimnames.

danerkestey committed:
Added Placeholder for predict.glm-like Function and Error Checks

Added a placeholder function structure similar to predict.glm with all parameters currently set to NULL. Error checks for matching dimensions implemented.

danerkestey commented:
Addressed comments
